src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -364,26 +377,70 @@ export class EditorPart extends Part implements IEditorPart {
 			return; // we have not been created yet
 		}
 
+		if (this.isGroupMaximized(target)) {
```

Comment:
Is there a way to have this in the `switch/case` or is that duplicating things?

Reply:
It would require duplicating things and does not make sense to be placed there. Arrangements that we add in the future will also rely on exiting the maximized editor first